### PR TITLE
Cast to Number instead of Double in ExtractAbbreviation.java (Emmet extension)

### PR DESCRIPTION
**Pull request by jonburs**
Fixes https://github.com/angelozerr/lsp4xml/issues/219

### CODE REVIEW

File: extensions/org.eclipse.lsp4xml.extensions.emmet/src/main/java/org/eclipse/lsp4xml/extensions/emmet/emmetio/extract/ExtractAbbreviation.java [hunk 1]
```diff
@@ -75,9 +75,9 @@ public static ExtractAbbreviationResult extractAbbreviation(String line, Integer
 				return null;
 			}
 			String abbreviation = (String) s.get("abbreviation");
-			Double location = (Double) s.get("location");
-			Double start = (Double) s.get("start");
-			Double end = (Double) s.get("end");
+			Number location = (Number) s.get("location");
```
Comment (angelozerr) [inline extensions/org.eclipse.lsp4xml.extensions.emmet/src/main/java/org/eclipse/lsp4xml/extensions/emmet/emmetio/extract/ExtractAbbreviation.java]: To support both JDK8 and JDK11, I suggets you to have a private  static method `getInt` which manages Number and Double like this:

```java
private static int getInt(Bindings bindings, String key) {
	Object result = bindings.get(key);		
	if (result instanceof Number) {
		return ((Number) result).intValue();
	}
	if (result instanceof Double) {
		return ((Double) result).intValue();
	}
	return 0;
}
```

After that you call it like this:

`int location = getInt(s, "location");`

Comment (jonburs) [reply extensions/org.eclipse.lsp4xml.extensions.emmet/src/main/java/org/eclipse/lsp4xml/extensions/emmet/emmetio/extract/ExtractAbbreviation.java]: I don't believe this is necessary. `Double` and `Integer` (et al.) both implement `Number` (see https://docs.oracle.com/javase/8/docs/api/java/lang/Number.html), so the current change works on both platforms -- were this not the case the CI run would have failed.

**Comment by angelozerr** [comment]
You are right! Please ignore my comment. I will merge your pr but emmet
support should ne rewritten in pure java to avoid using JS engine.

My idea is to provide an emmet support based on XML schema. I will describe
more my idea in an issue. Are you interested with our emmet support?

Le jeu. 11 juil. 2019 18:31, Jonathan Burstein <notifications@github.com> a
écrit :

**Comment by angelozerr** [comment]
Thanks @jonburs !

**Pull request merged**